matrix/README.md

```diff
@@ -0,0 +1,300 @@
+![charmory logo](../docs/assets/charmory.png)
+
+---
+
+# Overview
+
+Armory-matrix provides a utility to automatically expand parameter
+specifications in order to evaluate all combinations of parameters.
+
+For example, given two parameters `x` and `y` where `x` can be a number
+between 1 and 3 and `y` can be 1.5 or 6.5, we have the following matrix of
+parameter combinations:
+
+&nbsp; | x | y
+-------|---|--
+1      | 1 | 1.5
+2      | 1 | 6.5
+3      | 2 | 1.5
+4      | 2 | 6.5
+5      | 3 | 1.5
+6      | 3 | 6.5
+
+As the range of values for each parameter increases, and as more variable
+parameters are added, the matrix becomes significantly larger.
+
+# Installation
+
+```bash
+pip install armory-matrix
+```
+
+# Usage
+
+Use the `matrix` decorator to specify a set or range of values for parameters of
+a function. When the decorated function is invoked, it will automatically be
+called once for each combination of input parameters.
+
+```python
+from armory.matrix import matrix
+
+@matrix(
+    x=range(1, 3),
+    y=[1.5, 6.5],
+)
+def print_xy(x, y):
+    print(f"{x=}, {y=}")
+
+print_xy()
+```
+
+Will produce the following output:
+
+```
+x=1, y=1.5
+x=1, y=6.5
+x=2, y=1.5
+x=2, y=6.5
+```
+
+This is roughly equivalent to the following:
```

Comment:
This is equivalent to the nested loops below.

Reply:
I used "roughly" as that's used in the standard Python documentation for similar code snippets ([for example](https://docs.python.org/3/library/itertools.html#itertools.product)). Do you think it's unnecessary to differentiate that it's not exactly equivalent?